cmake/compiler-warnings.cmake

```diff
@@ -6,8 +6,14 @@ CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wall)
 
 CHECK_COMPILER_FLAG_AND_ENABLE_IT(-fno-strict-aliasing)
 
-CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wformat)
-CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wformat-security)
+if(WIN32)
+  # MSYS2 gcc compiler gives false positive warnings for (format (printf, 1, 2) - need to turn off for the time being
+  CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wno-format)
+
+  CHECK_COMPILER_FLAG_AND_ENABLE_IT(-mms-bitfields)
+else()
+  CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wformat-security)
```

Comment:
YES.
Now that it does `CHECK_COMPILER_FLAG_AND_ENABLE_IT(-Wall)`, it may be redundant, but better safe than sorry.
If you want, also look at https://redmine.darktable.org/attachments/2583/CMakeError.log#L49 (from https://redmine.darktable.org/issues/11199)